File: frontend/src/data/writers.js
export const writers = [
  {
    id: "xiong-mo",
    name: "墨寒",
    avatar: "🐺",
    style: "冷峻硬核",
    description: "擅长末日、废土、生存题材。文风冷峻克制，节奏快，动作描写扎实，对话精炼有力。",
    genres: ["末日", "废土", "科幻", "生存"],
    config: {
      writing_rules: "文风冷峻克制，禁止无意义抒情\n动作场景用短句快节奏，每句不超过15字\n对话不超过3句一组，穿插动作\n开篇直入冲突，不铺垫不寒暄",
      behavior_locks: "所有角色/不煽情不矫情，情绪通过动作和环境表达",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 2, max_inner_sentences: 2 }
    }
  },
  {
    id: "shui-yue",
    name: "水月",
    avatar: "🌙",
    style: "细腻温婉",
    description: "擅长言情、古代、宫廷题材。文笔细腻，情感描写深入，擅长营造氛围和心理刻画。",
    genres: ["言情", "古代", "宫廷", "都市"],
    config: {
      writing_rules: "情感描写要含蓄，用细节传递而非直说\n环境描写要有情绪映射（景随情移）\n对话要有潜台词，不在嘴边的话更重要",
      behavior_locks: "主角/不为爱情放弃底线和尊严",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 4, max_inner_sentences: 4 }
    }
  },
  {
    id: "po-feng",
    name: "破风",
    avatar: "⚡",
    style: "热血爽快",
    description: "擅长玄幻、修仙、都市异能。节奏极快，爽点密集，擅长打脸和升级的爽感设计。",
    genres: ["玄幻", "修仙", "都市异能", "系统"],
    config: {
      writing_rules: "每章至少一个爽点或转折，不能连续平淡\n打脸场景要写全：对手嚣张→主角出手→围观震惊→对手后悔\n升级/突破场景要有仪式感",
      behavior_locks: "主角/不圣母，不妇人之仁\n反派/被打脸后要有合理的后续反应",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 2, max_inner_sentences: 2 }
    }
  },
  {
    id: "qiu-shui",
    name: "秋水",
    avatar: "🍂",
    style: "悬疑烧脑",
    description: "擅长悬疑、推理、恐怖题材。擅长设伏笔和线索，节奏控制精准，擅长营造紧张氛围。",
    genres: ["悬疑", "推理", "恐怖", "惊悚"],
    config: {
      writing_rules: "每章至少埋一条伏笔或回收一条伏笔\n线索要公平呈现，不能事后补设定\n恐惧来自未知，不直接描写恐怖画面\n章末必须是悬念断章",
      behavior_locks: "所有角色/不主动暴露关键信息（除非有合理动机）\n作者/禁止在正文中暗示凶手身份",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 3, max_inner_sentences: 3 }
    }
  },
  {
    id: "yun-fei",
    name: "云飞",
    avatar: "☁️",
    style: "轻松幽默",
    description: "擅长都市、日常、轻喜剧。文风轻松，对话有趣，擅长搞笑桥段和日常生活的趣味描写。",
    genres: ["都市", "日常", "轻喜剧", "校园"],
    config: {
      writing_rules: "对话要有梗，每段对话后可以跟一句吐槽或内心OS\n搞笑节奏：铺垫→反差→吐槽，三拍子\n严肃场景不能持续超过500字，必须穿插轻松元素",
      behavior_locks: "主角/可以吐槽但不能刻薄",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 2, max_inner_sentences: 3 }
    }
  },
  {
    id: "long-ling",
    name: "龙鳞",
    avatar: "🐉",
    style: "史诗大气",
    description: "擅长历史、权谋、战争题材。文风大气磅礴，擅长群像描写和宏大场景，节奏沉稳有力。",
    genres: ["历史", "权谋", "战争", "架空"],
    config: {
      writing_rules: "场景描写要有画面感，像镜头语言一样推进\n多人场景要有层次，不能流水账\n权谋博弈要有信息差，读者要能跟上逻辑\n战争场面要有全局视角和局部细节交替",
      behavior_locks: "所有角色/行为必须有合理动机，不能为剧情强行行动",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 5, max_inner_sentences: 3 }
    }
  },
  {
    id: "su-qing",
    name: "苏青",
    avatar: "🌸",
    style: "甜宠治愈",
    description: "擅长甜宠、治愈、轻松恋爱。擅长甜蜜互动和温馨日常，让读者会心一笑。",
    genres: ["甜宠", "治愈", "恋爱", "日常"],
    config: {
      writing_rules: "甜宠互动要有心动感，描写要细腻但不腻\n每个甜蜜场景后可以跟一点小虐或误会增加张力\n角色互动要有化学反应，不能只有对话",
      behavior_locks: "主角/不恋爱脑到失去自我",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 2, max_inner_sentences: 3 }
    }
  },
  {
    id: "ye-feng",
    name: "夜风",
    avatar: "🌑",
    style: "黑暗写实",
    description: "擅长犯罪、现实主义、社会题材。文风压抑写实，擅长人性刻画和道德困境。",
    genres: ["犯罪", "现实主义", "社会", "人性"],
    config: {
      writing_rules: "不做道德审判，让读者自己判断角色行为\n现实逻辑优先，不开金手指\n灰色地带的角色最有魅力，不要非黑即白\n结局不必圆满，合理即可",
      behavior_locks: "作者/禁止在正文中评判角色对错\n所有角色/行为要符合其所处环境和立场",
      prohibited_words: "",
      advanced: { natural_dialogue: true, no_useless_details: true, max_env_sentences: 3, max_inner_sentences: 4 }
    }
  }
]

File: backend/novel/writers.py
WRITERS = [
    {
        "id": "xiong-mo",
        "name": "墨寒",
        "avatar": "🐺",
        "style": "冷峻硬核",
        "description": "擅长末日、废土、生存题材。文风冷峻克制，节奏快，动作描写扎实，对话精炼有力。",
        "genres": ["末日", "废土", "科幻", "生存"],
        "style_example": "“操。”赵小田骂了一句。\n面前这地，裂得跟龟壳似的，别说种庄稼，撒把种子都能掉缝里去。\n他蹲下来，抓了一把土，捏了捏，又闻了闻——碱的。\n“殿下，要不……咱们换个地方？”身后的老太监小心翼翼地开口。\n“换？”赵小田站起身，拍了拍手，“能换去哪儿？整个京城外头全是这鬼样子。”\n老太监缩了缩脖子，不敢吭声了。\n赵小田没理他，从怀里掏出一袋化肥。\n“就这儿了。”他说。",
        "config": {
            "writing_rules": "文风冷峻克制，禁止无意义抒情\n动作场景用短句快节奏，每句不超过15字\n对话不超过3句一组，穿插动作\n开篇直入冲突，不铺垫不寒暄",
            "behavior_locks": "所有角色/不煽情不矫情，情绪通过动作和环境表达",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 2, "max_inner_sentences": 2}
        }
    },
    {
        "id": "shui-yue",
        "name": "水月",
        "avatar": "🌙",
        "style": "细腻温婉",
        "description": "擅长言情、古代、宫廷题材。文笔细腻，情感描写深入，擅长营造氛围和心理刻画。",
        "genres": ["言情", "古代", "宫廷", "都市"],
        "style_example": "她没抬头。\n窗外的雨细细密密的，像是谁在天上拿筛子往下筛。茶已经凉了，她端着，也没喝。\n“娘娘……”\n“别说话。”\n宫女缩回手，退到帘子后面去了。\n她还记得他走的时候的样子——披风上的金线在阳光下闪了一下，然后人就不见了。\n三年了。\n茶凉了可以续，人呢？",
        "config": {
            "writing_rules": "情感描写要含蓄，用细节传递而非直说\n环境描写要有情绪映射（景随情移）\n对话要有潜台词，不在嘴边的话更重要",
            "behavior_locks": "主角/不为爱情放弃底线和尊严",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 4, "max_inner_sentences": 4}
        }
    },
    {
        "id": "po-feng",
        "name": "破风",
        "avatar": "⚡",
        "style": "热血爽快",
        "description": "擅长玄幻、修仙、都市异能。节奏极快，爽点密集，擅长打脸和升级的爽感设计。",
        "genres": ["玄幻", "修仙", "都市异能", "系统"],
        "style_example": "一拳。\n秦烈的拳头砸在对手脸上，血和牙齿一起飞出去。\n“还有谁？”\n擂台下面一片死寂。\n刚才还叫嚣着要让他好看的那几个人，现在低着头，连大气都不敢出。\n“没人了？”秦烈甩了甩手上的血，“那我走了。”\n他转身，台下这才炸开——\n“卧槽！三连胜！”\n“这小子吃了什么药？”\n“不是吃药，是疯了……”",
        "config": {
            "writing_rules": "每章至少一个爽点或转折，不能连续平淡\n打脸场景要写全：对手嚣张→主角出手→围观震惊→对手后悔\n升级/突破场景要有仪式感",
            "behavior_locks": "主角/不圣母，不妇人之仁\n反派/被打脸后要有合理的后续反应",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 2, "max_inner_sentences": 2}
        }
    },
    {
        "id": "qiu-shui",
        "name": "秋水",
        "avatar": "🍂",
        "style": "悬疑烧脑",
        "description": "擅长悬疑、推理、恐怖题材。擅长设伏笔和线索，节奏控制精准，擅长营造紧张氛围。",
        "genres": ["悬疑", "推理", "恐怖", "惊悚"],
        "style_example": "第三个被害者出现的时候，林深终于意识到这不是普通的连环杀人案。\n前两起案子互相之间没有任何联系——不同的城市、不同的作案手法、不同的被害者类型。\n唯一的共同点是：被害者的口袋里都有一颗白色棋子。\n围棋的白子。\n林深把那颗棋子捏在手里，翻来覆去地看。\n“队长，报告出来了。”同事递过来一份文件。\n他没有接。\n“被害者的人际关系查过了？”\n“查了，没有交集。”\n“那棋子呢？”\n“普通的云子，店里到处都能买到。”\n林深把棋子放在桌上，突然问了一句：\n“三颗棋子，分别在两个省三个市的三个案发现场——凶手怎么保证我们一定会注意到它们？”\n办公室里安静了三秒。",
        "config": {
            "writing_rules": "每章至少埋一条伏笔或回收一条伏笔\n线索要公平呈现，不能事后补设定\n恐惧来自未知，不直接描写恐怖画面\n章末必须是悬念断章",
            "behavior_locks": "所有角色/不主动暴露关键信息（除非有合理动机）\n作者/禁止在正文中暗示凶手身份",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 3, "max_inner_sentences": 3}
        }
    },
    {
        "id": "yun-fei",
        "name": "云飞",
        "avatar": "☁️",
        "style": "轻松幽默",
        "description": "擅长都市、日常、轻喜剧。文风轻松，对话有趣，擅长搞笑桥段和日常生活的趣味描写。",
        "genres": ["都市", "日常", "轻喜剧", "校园"],
        "style_example": "“你说什么？”\n“我说……要不咱俩凑合凑合？”\n林小晚盯着面前这个一脸真诚的男生，沉默了三秒。\n“你是在跟我求婚？”\n“不是不是不是！”男生疯狂摆手，“我是说，搭伙过日子。房租平摊，水电平摊，饭我做。”\n“……”\n“你负责吃就行。”\n林小晚想了想：“你会做什么菜？”\n“番茄炒蛋。”\n“还有呢？”\n“……番茄炒蛋。”\n“滚。”",
        "config": {
            "writing_rules": "对话要有梗，每段对话后可以跟一句吐槽或内心OS\n搞笑节奏：铺垫→反差→吐槽，三拍子\n严肃场景不能持续超过500字，必须穿插轻松元素",
            "behavior_locks": "主角/可以吐槽但不能刻薄",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 2, "max_inner_sentences": 3}
        }
    },
    {
        "id": "long-ling",
        "name": "龙鳞",
        "avatar": "🐉",
        "style": "史诗大气",
        "description": "擅长历史、权谋、战争题材。文风大气磅礴，擅长群像描写和宏大场景，节奏沉稳有力。",
        "genres": ["历史", "权谋", "战争", "架空"],
        "style_example": "大军压境的时候，城墙上只有三千人。\n三千对三万。\n副将的脸色白得像纸，握着剑柄的手指节发青。“将军……”\n沈不言没回头，他的目光越过城墙，落在远处漫天的烟尘里。\n“知道为什么我一直不调兵吗？”\n“末将不知。”\n“因为调了也没用。”沈不言笑了一下，那笑容里没有恐惧，“京城空虚，边疆告急，朝堂上那些人巴不得我死在这里。”\n他拍了拍城墙上的砖。\n“所以他们不知道——我从来没打算守住这座城。”\n副将愣住了。\n“我要的是他们都以为我会守城。”",
        "config": {
            "writing_rules": "场景描写要有画面感，像镜头语言一样推进\n多人场景要有层次，不能流水账\n权谋博弈要有信息差，读者要能跟上逻辑\n战争场面要有全局视角和局部细节交替",
            "behavior_locks": "所有角色/行为必须有合理动机，不能为剧情强行行动",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 5, "max_inner_sentences": 3}
        }
    },
    {
        "id": "su-qing",
        "name": "苏青",
        "avatar": "🌸",
        "style": "甜宠治愈",
        "description": "擅长甜宠、治愈、轻松恋爱。擅长甜蜜互动和温馨日常，让读者会心一笑。",
        "genres": ["甜宠", "治愈", "恋爱", "日常"],
        "style_example": "她把奶茶递过去，他没接。\n“怎么了？”她问。\n“你加冰了。”\n“你不喝加冰的？”\n他叹了口气，接过奶茶捂在手心里。“不是不喝。”\n那算了——她正准备说。\n“是你来例假了，别喝凉的。”\n她愣住。\n他怎么知道的？\n……等等。\n“你……”她脸一下子红了，“你算我日子？！”\n他别过头，耳朵尖红的。",
        "config": {
            "writing_rules": "甜宠互动要有心动感，描写要细腻但不腻\n每个甜蜜场景后可以跟一点小虐或误会增加张力\n角色互动要有化学反应，不能只有对话",
            "behavior_locks": "主角/不恋爱脑到失去自我",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 2, "max_inner_sentences": 3}
        }
    },
    {
        "id": "ye-feng",
        "name": "夜风",
        "avatar": "🌑",
        "style": "黑暗写实",
        "description": "擅长犯罪、现实主义、社会题材。文风压抑写实，擅长人性刻画和道德困境。",
        "genres": ["犯罪", "现实主义", "社会", "人性"],
        "style_example": "审讯室的灯很亮。\n亮得让人睁不开眼。\n“你再说一遍。”警察的声音不大，但压得很低。\n“我说了，是我干的。”\n对面的男人三十出头，穿着一件洗得发白的夹克，双手放在桌上，手腕上没戴铐子。\n他的表情很平静。\n“为什么？”\n“没有为什么。”\n“你认识被害者吗？”\n“不认识。”\n“那为什么杀他？”\n男人沉默了很久。久到警察以为他不会回答了。\n“因为他活着，我女儿就活不了。”",
        "config": {
            "writing_rules": "不做道德审判，让读者自己判断角色行为\n现实逻辑优先，不开金手指\n灰色地带的角色最有魅力，不要非黑即白\n结局不必圆满，合理即可",
            "behavior_locks": "作者/禁止在正文中评判角色对错\n所有角色/行为要符合其所处环境和立场",
            "prohibited_words": "",
            "advanced": {"natural_dialogue": True, "no_useless_details": True, "max_env_sentences": 3, "max_inner_sentences": 4}
        }
    }
]

WRITERS_MAP = {w["id"]: w for w in WRITERS}
